commit f78f501857cec8bb682b76faa9cbd5cc1e0dcad4
Author: Willy Tarreau <w@1wt.eu>
Date:   Fri Mar 29 18:56:54 2019 +0100

    BUILD: Makefile: disable shared cache on AIX 5.1
    
    AIX 5.1 is missing the following builtins used for atomic locking of the
    shared inter-process cache :
    
       .__sync_val_compare_and_swap_4
       .__sync_lock_test_and_set_4
       .__sync_sub_and_fetch_4
    
    Let's simply use the private cache by default since nobody cares on
    such old systems. No test was made on a more recent version.
    
    (cherry picked from commit 13d9b0231abe6e1d6c404725746308a98bdab8c6)
    Signed-off-by: Willy Tarreau <w@1wt.eu>
    (cherry picked from commit 3349375d3f5820f261f9c3058ca6e5ced67f3505)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/Makefile b/Makefile
index 955f1188..36df7b29 100644
--- a/Makefile
+++ b/Makefile
@@ -358,7 +358,8 @@ ifeq ($(TARGET),aix51)
   # This is for AIX 5.1
   USE_POLL        = implicit
   USE_LIBCRYPT    = implicit
-  TARGET_CFLAGS   = -Dss_family=__ss_family -Dip6_hdr=ip6hdr -DSTEVENS_API -D_LINUX_SOURCE_COMPAT
+  USE_PRIVATE_CACHE = implicit
+  TARGET_CFLAGS   = -Dss_family=__ss_family -Dip6_hdr=ip6hdr -DSTEVENS_API -D_LINUX_SOURCE_COMPAT -Dunsetenv=my_unsetenv
   DEBUG_CFLAGS    =
 else
 ifeq ($(TARGET),aix52)
